### PR TITLE
fix mcl import

### PR DESCRIPTION
**Pull request by iulianpascalau**
fixed mcl init

### CODE REVIEW

File: crypto/signing/mcl/suiteBLS12_381.go [hunk 1]
```diff
@@ -41,12 +41,13 @@ func init() {
 
 	pubKey := &bls.PublicKey{}
 	bls.BlsGetGeneratorForPublicKey(pubKey)
+	blsSwapG := bls.IsSwapG()
 	if blsSwapG {
-		generatorG1 := bls.CastG1FromPublicKey(pubKey)
+		generatorG1 := (*bls.G1)(unsafe.Pointer(pubKey))
```
Comment (AdoAdoAdo) [inline crypto/signing/mcl/suiteBLS12_381.go]: please use the casting functions provided

File: crypto/signing/mcl/suiteBLS12_381.go [hunk 2]
```diff
@@ -41,12 +41,13 @@ func init() {
...
 		basePointG1Str.Store(generatorG1.GetString(10))
 		basePointG2Str.Store(g2str)
 	} else {
-		generatorG2 := bls.CastG2FromPublicKey(pubKey)
+		generatorG2 := (*bls.G2)(unsafe.Pointer(pubKey))
```
Comment (AdoAdoAdo) [inline crypto/signing/mcl/suiteBLS12_381.go]: please use the casting function provided

File: crypto/signing/mcl/suiteBLS12_381.go (hunk 1 above)
Comment (iulianpascalau) [reply crypto/signing/mcl/suiteBLS12_381.go]: cast calls bls.IsSwapG(). Added TODO for a later fix

File: crypto/signing/mcl/suiteBLS12_381.go (hunk 2 above)
Comment (iulianpascalau) [reply crypto/signing/mcl/suiteBLS12_381.go]: cast calls bls.IsSwapG(). Added TODO for a later fix

**Pull request merged**